pubsub/pulsar/pulsar.go

```diff
@@ -123,16 +123,30 @@ func (p *Pulsar) Publish(req *pubsub.PublishRequest) error {
 		producer = cache.(pulsar.Producer)
 	}
 
-	_, err = producer.Send(context.Background(), &pulsar.ProducerMessage{
-		Payload: req.Data,
-	})
-	if err != nil {
+	msg := parsePublishMetadata(req)
+	if _, err = producer.Send(context.Background(), msg); err != nil {
 		return err
 	}
 
 	return nil
 }
 
+// parsePublishMetadata parse publish metadata
+func parsePublishMetadata(req *pubsub.PublishRequest) (msg *pulsar.ProducerMessage) {
+	msg = &pulsar.ProducerMessage{
+		Payload: req.Data,
+	}
+	if val, ok := req.Metadata["deliver_at"]; ok {
+		msg.DeliverAt, _ = time.Parse("2006-01-02 15:04:05", val)
```

Comment:
Is there a reason why this needs a custom time format? If not, I would recommend standardizing on the `time.RFC3339` format as most existing components use.

Also, do not ignore the parse error, the method should fail with an error on invalid inputs.